packages/insured-bridge-relayer/src/Relayer.ts

```diff
@@ -96,228 +167,179 @@ export class Relayer {
-            message: "Relay pending and already sped up 😖",
-            realizedLpFeePct: realizedLpFeePct.toString(),
-            relayState: relayableDeposit.status,
-            hasInstantRelayer,
-            relayableDeposit,
+            message: "Unexpected error processing deposit",
+            error,
           });
-          continue;
         }
-        const shouldRelay = await this.shouldRelay(
-          relayableDeposit.deposit,
-          relayableDeposit.status,
-          realizedLpFeePct,
-          hasInstantRelayer
-        );
-
-        // Depending on value of `shouldRelay`, send correct type of relay.
-        await this.sendRelayTransaction(
-          shouldRelay,
-          realizedLpFeePct,
-          relayableDeposit,
-          pendingRelay,
-          hasInstantRelayer
-        );
       }
     }
 
     return;
   }
 
-  async checkForPendingRelaysAndDispute(): Promise<void> {
-    this.logger.debug({ at: "InsuredBridgeRelayer#Disputer", message: "Checking for pending relays and disputing" });
+  // Evaluates given pending `relay` and determines whether to submit a dispute.
+  private async _disputePendingRelay(relay: Relay): Promise<void> {
+    // Check if relay has expired, in which case we cannot dispute.
+    const relayExpired = await this.isRelayExpired(relay, relay.l1Token);
+    if (relayExpired.isExpired) {
+      this.logger.debug({
+        at: "InsuredBridgeRelayer#Disputer",
+        message: "Pending relay has expired, ignoring",
+        relay,
+        expirationTime: relayExpired.expirationTime,
+        contractTime: relayExpired.contractTime,
+      });
+      return;
+    }
 
-    // Build dictionary of pending relays keyed by l1 token and deposit hash. We assume that getPendingRelays() filters
-    // out Finalized relays.
-    const pendingRelays: Relay[] = this.getPendingRelays();
-    if (pendingRelays.length == 0) {
-      this.logger.debug({ at: "InsuredBridgeRelayer#Disputer", message: "No pending relays" });
+    // If relay's chain ID is not whitelisted then dispute it.
+    if (!this.whitelistedChainIds.includes(relay.chainId)) {
+      this.logger.debug({
+        at: "InsuredBridgeRelayer#Disputer",
+        message: "Disputing pending relay with non-whitelisted chainID",
+        relay,
+      });
+      await this.disputeRelay(
+        {
+          chainId: relay.chainId,
+          depositId: relay.depositId,
+          depositHash: relay.depositHash,
+          l1Recipient: relay.l1Recipient,
+          l2Sender: relay.l2Sender,
+          l1Token: relay.l1Token,
+          amount: relay.amount,
+          slowRelayFeePct: relay.slowRelayFeePct,
+          instantRelayFeePct: relay.instantRelayFeePct,
+          quoteTimestamp: relay.quoteTimestamp,
+          depositContract: (await this.l1Client.bridgeAdmin.methods.depositContracts(relay.chainId).call())[0],
+          // `depositContracts()` returns [depositContract, messengerContract] and we want the first arg.
```

Comment:
can we remove this?

Reply:
hmmm I think this comment is worth keeping